specs/eventing/data-plane.md

```diff
@@ -1,125 +1,179 @@
-MUST respond with `400 Bad Request`.
-
-### Content Modes Supported
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in RFC2119.
 
-A Sink MUST support `Binary Content Mode` and `Structured Content Mode` as
-described in
-[HTTP Message Mapping section of HTTP Protocol Binding for CloudEvents](https://github.com/cloudevents/spec/blob/master/http-protocol-binding.md#3-http-message-mapping)
+When not specified in this document, the
+[CloudEvents HTTP bindings, version 1.0](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md)
+and [HTTP 1.1 protocol](https://tools.ietf.org/html/rfc7230) standards should be
+followed (with the CloudEvents bindings preferred in the case of conflict).
 
-A Sink MAY support `Batched Content Mode` but that mode is not used in Knative
-Eventing currently (that may change in future).
+The current version of this document does not describe protocol negotiation or
+any delivery mechanism other than HTTP 1.1. Future versions may define protocol
+negotiation to optimize delivery; compliant implementations SHOULD aim to
+interoperate by ignoring unrecognized negotiation options (such as
+[HTTP `Upgrade` headers](https://datatracker.ietf.org/doc/html/rfc7230#section-6.7)).
 
-### Retries
+## Event Delivery
 
-Sinks should expect that retries and accept possibility that duplicate events
-may be delivered.
+To provide simpler support for event sources which might be translating events
+from existing systems, some data plane requirements for senders are relaxed in
+the general case. In the case of Knative Eventing provided resources (Channels
+and Brokers) which implement these roles, requirements are increased from SHOULD
+to MUST. Exceptions with stronger requirements for Channels and Brokers are
+called out as they occur.
 
-### Error handling
+### Minimum supported protocol
 
-If Sink is not returning HTTP success header (200 or 202) then the event may be
-sent again. If the event can not be delivered then some sources of events (such
-as Knative sources, brokers or channels) MAY support
-[dead letter sink or channel](https://github.com/knative/eventing/blob/main/docs/delivery/README.md) for events that can not be
-delivered.
+All senders and recipients MUST support the CloudEvents 1.0 protocol and the
+[binary](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#31-binary-content-mode)
+and
+[structured](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#32-structured-content-mode)
+content modes of the CloudEvents HTTP binding. Senders which do not advertise
+the ability to accept [reply events](#derived-reply-events) MAY implement only
+one content mode, as the recipient is not allowed to negotiate the content mode.
+Senders MUST support both cleartext (`http`) and TLS (`https`) URLs as event
```

Comment:
Are "senders" event producers or just any component within the Knative OSS project that sends events - like "adapters"? I think we need a terminology section because if "sender" == "producer" then I think requiring http is a show stopper for some event producers? If a system rejects a CR with a destination value of "http" because they want to mandate https, are they in violation of this MUST?